x-pack/plugins/gis/server/maps_telemetry/maps_telemetry.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+
+function getSavedObjectsClient(server, callCluster) {
+  const { SavedObjectsClient, getSavedObjectsRepository } = server.savedObjects;
+  const internalRepository = getSavedObjectsRepository(callCluster);
+  return new SavedObjectsClient(internalRepository);
+}
+
+function getUniqueLayerCounts(layerCountsList, mapsCount) {
+  const uniqueLayerTypes = _.uniq(_.flatten(
+    layerCountsList.map(lTypes => Object.keys(lTypes))));
+
+  return uniqueLayerTypes.reduce((accu, type) => {
+    const typeCounts = layerCountsList.reduce((accu, tCounts) => {
+      tCounts[type] && accu.push(tCounts[type]);
+      return accu;
+    }, []);
+    accu[type] = {
+      min: _.min(typeCounts),
+      max: _.max(typeCounts),
+      avg: _.sum(typeCounts) / mapsCount
+    };
+    return accu;
+  }, {});
+}
+
+async function buildMapsTelemetry(savedObjectsClient) {
+  const gisMapsSavedObject = await savedObjectsClient.find({
+    type: 'gis-map'
+  });
+
+  const savedMaps = _.get(gisMapsSavedObject, 'saved_objects')
+    .map(savedMapObject => JSON.parse(savedMapObject.attributes.layerListJSON));
+  const mapsCount = savedMaps.length;
+
+  const dataSourcesCount = savedMaps.map(lList => {
+    const sourceIdList = lList.map(layer => layer.sourceDescriptor.id);
+    return _.uniq(sourceIdList).length;
+  });
+
+  const layersCount = savedMaps.map(lList => lList.length);
+  const layerTypesCount = savedMaps.map(lList => _.countBy(lList, 'type'));
+
+  // Count of EMS Vector layers used
+  const emsLayersCount = savedMaps.map(lList => _(lList)
+    .countBy(layer => {
+      const isEmsFile = _.get(layer, 'sourceDescriptor.type') === 'EMS_FILE';
```

Comment:
extract this constant `EMS_FILE` to constant (e.g. in `common/constants.js), so it can be referenced from single spot (right now, constant is defined here: https://github.com/thomasneirynck/kibana/blob/9520a812981bf58f37dfa40758af6e99d1af6a0f/x-pack/plugins/gis/public/shared/layers/sources/ems_file_source/ems_file_source.js#L16)